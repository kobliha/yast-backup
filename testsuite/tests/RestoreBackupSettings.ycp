// Backup module - testsuite
//
// RestoreDefaultSettings function test
//
// $Id$
//

{

include "testsuite.ycp";

map readmap = $[ "target" : $["string" : "", "size" : -1, "tmpdir" : "/tmp" ],
    "proc" : $["cpuinfo" : $[ "value" : $["0" : $[ "flags" : ""]]]],
    "probe" : $["has_smp" : false, "cdrom" : $["manual" : []]],
    "sysconfig" : $[ "autoinstall" : $[
      "REPOSITORY" : "/var/lib/autoinstall/repository",
      "ALICE_REPOSITORY" : "/var/lib/autoinstall/repository",
      "CLASS_DIR" : "/var/lib/autoinstall/repository/classes",
      "XSLT_DONTMERGE" : "addon,conf"
    ]]
    ];

map execmap = $[ "target" : $["bash_output" : $["exit":0, "stderr":"", "stdout":"charmap=\"ISO-8859-1\"\n"]]];

TESTSUITE_INIT([ readmap, $[], execmap ], nil);

import "Pkg";   // override packamanager
import "Backup";

DUMP( Backup::backup_profiles);

    // setup global settings
    Backup::archive_name = "Does not work";
    Backup::description = "Does not work - profile";
    Backup::archive_type = nil;
    Backup::multi_volume = nil;
    Backup::volume_size = nil;
    Backup::user_volume_size = nil;
    Backup::user_volume_unit = nil;
    Backup::search = nil;
    Backup::system = nil;
    Backup::display = nil;
    Backup::do_md5_test = nil;
    Backup::default_dir = nil;
    Backup::dir_list = nil;
    Backup::fs_exclude = nil;
    Backup::detected_fs = nil;
    Backup::detected_ext2 = nil;
    Backup::ext2_backup = nil;
    Backup::backup_pt = nil;
    Backup::backup_all_ext2 = nil;
    Backup::backup_none_ext2 = nil;
    Backup::backup_selected_ext2 = nil;
    Backup::unselected_files = nil;

    Backup::selected_files = nil;
    Backup::backup_files = $[ ];

Backup::RestoreDefaultSettings();

Backup::StoreSettingsToBackupProfile( "test profile" );

DUMP( Backup::backup_profiles);

}

