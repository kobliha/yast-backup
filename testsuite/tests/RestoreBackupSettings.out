Dump	$[]
Dump	$["test profile":$[`archive_name:"", `archive_type:`tgz, `autoprofile:true, `backup_all_ext2:false, `backup_none_ext2:true, `backup_pt:true, `backup_selected_ext2:false, `default_dir:["/media", "/tmp", "/var/lock", "/var/run", "/var/tmp", "/var/cache", "/sys", "/windows", "/mnt"], `description:"", `detected_ext2:nil, `detected_fs:nil, `dir_list:["/media", "/tmp", "/var/lock", "/var/run", "/var/tmp", "/var/cache", "/sys", "/windows", "/mnt"], `display:false, `do_md5_test:true, `ext2_backup:[], `fs_exclude:["iso9660", "ntfs"], `include_dirs:["/"], `mail_summary:true, `multi_volume:false, `nfsexport:"", `nfsserver:"", `perms:true, `regexp_list:[], `search:true, `system:false, `target_type:`file, `tmp_dir:"/tmp", `unselected_files:[], `user_volume_size:"", `user_volume_unit:nil, `volume_size:`fd144]]
